pkg/ingester/owned_streams.go

```diff
@@ -11,11 +11,11 @@ import (
 	"github.com/grafana/loki/v3/pkg/util/constants"
 )
 
```

Comment:
Probably worth adding a metric here to track the duration for recalculation.

Reply:
done